### PR TITLE
[Fixes #587] Splash screen on first load

### LINKED ISSUE #587: Welcome dialog on first load
With the Auction House revamp it was necessary to revamp Auctionator as well. It has been overhauled to work in 8.3

This means that things may not be where they were in 8.2 and features (most notably undercut scanning and price histories) missing.

You _do_ have all your old shopping lists and can edit them. All the old search functionality for shopping lists is back, just press "Search"

### PR DESCRIPTION
**Pull request by jrob8577**
* Option to turn off instead of only loading on first load
* Formats previously viewed version history differently (greyed out)
* wip: needs some formatting work for links, if possible?
* wip: needs localization

Initial view of unseen version
Second view
New version view

### CODE REVIEW

File: Source/SplashScreen/Mixins/Main.lua [hunk 1]
```diff
@@ -0,0 +1,157 @@
+-- TODO All this needs Localization
+-- We should probably move this into another file too
+local MESSAGES = {
+  {
+    Version = "8.3.0",
```
Comment (plusmouse) [inline Source/SplashScreen/Mixins/Main.lua:5]: To localise this we need to replace every string with an AUCTIONATOR_L_SPLASH_830_DESCRIPTION, AUCTIONATOR_L_SPLASH_830_SECTION_BUGS, AUCTIONATOR_L_SPLASH_830_BUGS_LINE_1, etc

File: Source/SplashScreen/Frames/Main.xml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        <Anchors>
+          <Anchor point="TOPLEFT" relativeKey="$parent.Inset" relativePoint="BOTTOMLEFT" x="-20" y="2"/>
+        </Anchors>
+      </Frame>
+    </Frames>
+
+    <Layers>
+      <Layer>
+        <!-- Title String TODO Needs Localization -->
+        <FontString text="Welcome to Auctionator TODO VERSION" inherits="GameFontHighlight">
```
Comment (plusmouse) [inline Source/SplashScreen/Frames/Main.xml:53]: If we omit the version from the string this is easy to localize

File: Source/Core/Mixin.lua [hunk 3]
```diff
@@ -78,6 +78,32 @@ local function InitializeAuctionHouseTabs()
   end
 end
 
+local function InitializeSplashScreen()
+  -- TODO Check for display setting before creating
```
Comment (plusmouse) [inline Source/Core/Mixin.lua:82]: Can't you just move the code that checks to the start of the function?

**Comment by plusmouse** [comment]
The splash screen should close when "Escape" is pressed

**Comment by flow0284** [comment]
German localization done and waiting ;)

File: Source/Core/Mixin.lua (hunk 3 above)
Comment (jrob8577) [reply Source/Core/Mixin.lua:82]: I need the reference to the frame for some of the functionality, unless I move that into global methods

**Pull request merged**